### PR TITLE
[DevHub Landing] Add logged-in state for mobile navigation.

### LINKED ISSUE #4027: [DevHub Landing] Navigation logged-in state
Supports mozilla/addons-frontend#1021

### PR DESCRIPTION
**Pull request by EnTeQuAk**
Fixes #4027

Since there isn't a mobile-version of the logged-in state for the navigation this is a bit guessing of what was ment to be seen. May need some input from @designakt or @brampitoyo to finalize. see the screenshot below for how it looks.

Mobile (avatar hidden currently)
screenshot from 2016-11-23 15-19-28

Tablet and larger

screenshot from 2016-11-23 15-20-31

### CODE REVIEW

File: src/olympia/devhub/templates/devhub/new-landing/index.html [hunk 1]
```diff
@@ -37,26 +37,31 @@
...
 
-    <nav class="DevHub-Navigation">
+    <nav class="DevHub-Navigation {% if request.user.is_authenticated() %}DevHub-Navigation--signedin{% endif %}">
       <ul>
         <li><a href="#"><div class="Logo" alt="{{ dev_page_title() }}" /></a></li>
         <li><a href="#">Docs</a></li>
         <li><a href="#">Support</a></li>
         <li><a href="#">Blog</a></li>
+        {% if request.user.is_authenticated() %}
+          <li><a href="#"><img class="avatar" src="{{ request.user.picture_url }}" /></a></li>
```
Comment (muffinresearch) [inline src/olympia/devhub/templates/devhub/new-landing/index.html]: One thing to bear in mind here is there's no concept of a mobile user's settings page (as far as the mobile pages v1 is concerned). So this link should probably log people out. That can be a separate issue of course.

Comment (EnTeQuAk) [inline src/olympia/devhub/templates/devhub/new-landing/index.html]: Good point, maybe it makes sense to rename that link just to "Logout" then.

File: static/css/devhub/base.less [hunk 2]
```diff
@@ -105,6 +108,29 @@ section {
...
+
+  ul {
+    li {
+      margin: 10px 10px;
+
+      a {
+        color: @secondary-font-color;
+      }
+
+      .Logo {
```
Comment (muffinresearch) [inline static/css/devhub/base.less]: This looks a bit over nested - this is going to result in some pretty long selectors e.g:

`.DevHub-Navigation--signedin ul li .Logo`

See if you can extract the Logo bit out to be on its own. Generally if nesting is more than 2 levels deep it's worth looking at again.

**Comment by muffinresearch** [comment]
From the code perspective r+wc 

Looks like circle is having some problems re: rpm signatures and the hashing issue is still there. Maybe rebasing will help since that the package hash fix landed already afaik.

**Comment by muffinresearch** [comment]
UX are going to look at the button but currently we need to also update this for the mobile pages since we don't have anything that reflects the current user presently. If whatever is decided can work in both places we should file an issue on addons-frontend to update it.

add-ons_for_firefox

**Comment by EnTeQuAk** [comment]
Alrighty, we're hiding the avatar for mobile for now and only show it on bigger tablet-ish devices.

**Comment by brampitoyo** [comment]
@EnTeQuAk I purposefully didn’t design a narrow layout for the signed-in state because the management interface isn’t fully functional there. Meaning, if you’re not using your desktop, you can’t upload a new add-on, a new version or a new file for a platform.

But a large majority of the management functions other than uploading file – changing description, listing/hiding, removing reviews, etc. – are things that should be able to be accomplished on mobile! And now that we’ve locked down what the layout looks like, on the main page of signed-out and signed-in states, we can apply it to the rest of the dev dashboard.

I suppose, my concern is, we should probably look into rethinking the devhub’s flow and functionalities in a more comprehensive redesign, instead of trying to fit all the old functionalities into the new CSS.

What are your thoughts? @designakt, @pwalm and @brassy-’s feedback may be useful here.

**Comment by EnTeQuAk** [comment]
I agree, I didn't intend this to be trying to force things into the new layout, just making sure that the new developers landing page works logged-in and logged-out. If you navigate away from this page you might end up on a desktop-only page anyway so that's expected for now. There's only a stretch goal to apply the navigation to more sections of the devhub if that's possible and reasonable.

For all the other functions you mentioned, yes, that's a whole other project imho and should be part of `addons-frontend` future work.

So more or less, the question here is, does the way I did the mobile-version of the signed-in state work for you? I now intentionally removed user-focused links from it, just changed the design to make it responsive to the desktop-version.

**Comment by designakt** [comment]
It is ok if the log-in state only shows your add-ons and tells you that for more you have to use the desktop-version. And it would be great if, as @muffinresearch mentioned, the Log-in, Log-out control would be the same on both systems. (That would probably entail some rework on the dev-hub side to match @pwalm layout for AMO) Phil, can you make a suggestion of how that should look on dev-hub.

**Pull request merged**